src/include/sof/audio/component.h

```diff
@@ -208,6 +208,7 @@ struct comp_ops {
  */
 struct comp_driver {
 	uint32_t type;		/**< SOF_COMP_ for driver */
+	uint16_t flavour;	/**< flavour for each comp type */
```

Comment:
This could break alignment for everything else following flavour if compiler packs by default, no harm in using a uint32_t even if IPC uses a uint16_t

Reply:
@lgirdwood good point. i'll fix it. But @plbossart has voiced a concern about the whole idea of flavours. So maybe something to align on first